lib/useSearchParams.ts

```diff
@@ -0,0 +1,24 @@
+import { useCallback, useMemo } from 'react';
+import { useRouter } from 'next/router';
+import { ParsedUrlQuery, parse } from 'querystring';
+
+export default function useSearchParams(asPath: string) {
+  const { push } = useRouter();
+  const { pathname, query } = useMemo(() => {
+    const delimiter = asPath.search('\\?');
+    return delimiter >= 0
+      ? {
+          pathname: asPath.substring(0, delimiter),
+          query: parse(asPath.substring(delimiter + 1)),
+        }
+      : { pathname: asPath, query: parse('') };
+  }, [asPath]);
+  const setSearchParams = useCallback(
+    (query: ParsedUrlQuery) => {
+      console.log(`\n\n\n\t\t${pathname}\n${query}\n\n\n`);
+      push({ pathname, query });
+    },
+    [pathname, push]
+  );
+  return [query, setSearchParams] as const;
```

Comment:
Why don't you use a simple way to fetch query params 

`const {query} = useRouter();`

and, Actually there is a good hook for this case. ([use-query-params](https://github.com/pbeshai/use-query-params))

Reply:
I wanted to stick with the tools provided by Next.js, and as I'm actually using the Next.js router, I think it's going to be a little difficult trying to use use-query-params as it seems like it's using React Router (I'm not sure if it's compatible with Next.js 😅)

Reply:
:+1: You can consider it next time, it is working perfectly with Next & Router.